cpp/src/gandiva/gdv_function_stubs.cc

```diff
@@ -794,6 +796,60 @@ const char* gdv_fn_initcap_utf8(int64_t context, const char* data, int32_t data_
   *out_len = out_idx;
   return out;
 }
+
+GANDIVA_EXPORT
+const char* gdv_fn_aes_encrypt(int64_t context, const char* data, int32_t data_len,
+                               const char* key_data, int32_t key_data_len,
+                               int32_t* out_len) {
+  if (data_len < 0) {
+    gdv_fn_context_set_error_msg(context, "Invalid data length to be encrypted");
+    *out_len = 0;
+    return "";
+  }
+
+  unsigned char encrypted[256];
```

Comment:
Why do we need this temporary buffer? Why not allocate memory first and pass a pointer to that buffer to aes_encrypt?

Reply:
We dont need it anymore